### PR TITLE
Intelligently read in fits files

### PR DESCRIPTION
**Pull request by nksaunders**
#316: Make sure that fits files are read in as the correct type. Add a `targetpixelfile.open()` function to check Kepler or TESS specific keywords and correspondingly open either a `KeplerTargetPixelFile` or `TessTargetPixelFile`.

### CODE REVIEW

**Comment by nksaunders** [comment]
```python
from lightkurve.targetpixelfile import open_fits

open_fits('/path/to/kepler/or/k2.fits') # returns a `KeplerTargetPixelFile`
open_fits('/path/to/tess.fits') # returns a `TessTargetPixelFile`
```

File: lightkurve/targetpixelfile.py [hunk 1]
```diff
@@ -1389,3 +1413,19 @@ def get_bkg_lightcurve(self, aperture_mask=None):
                               flux=np.nansum(self.flux_bkg[:, aperture_mask], axis=1),
                               flux_err=flux_bkg_err,
                               **keys)
+
+def open_fits(path):
```
Comment (barentsen) [inline lightkurve/targetpixelfile.py]: I think we can simply call it `open()`!  Can anyone think of a reason to call it `open_fits()`?

Comment (nksaunders) [reply lightkurve/targetpixelfile.py]: I opted for `open_fits()` over `open()` to preserve the built-in python function `open()`. But if you think this won't be an issue I'm happy to simplify it!

In that case, it should probably throw an exception if `path` doesn't point to a fits file.

Comment (barentsen) [reply lightkurve/targetpixelfile.py]: I think I prefer `lightkurve.targetpixelfile.open()` for consistency with `astropy.io.fits.open()`.

**Comment by barentsen** [comment]
@nksaunders Do you think we should extend `open()` to also detect and read in `LightCurveFile` files?

**Comment by nksaunders** [comment]
@barentsen `open()` now detects `TargetPixelFiles` and `LightCurveFiles` based on header information and returns the corresponding object.

I wasn't sure where to put this function, so I moved it to the `search` module for now. I initially thought it would make sense in `utils`, but as both `lightcurvefile` and `targetpixelfile` depend on `utils`, it created circular imports when I tried to return the correct object. Any thoughts on where it should end up?

File: lightkurve/tests/test_search.py [hunk 2]
```diff
@@ -125,3 +127,17 @@ def test_source_confusion():
     desired_target = 6507433
     tpf = search_targetpixelfile(desired_target, quarter=8).download()
     assert tpf.targetid == desired_target
+
+def test_open():
+    k2_path = os.path.dirname(os.path.abspath(__file__)) + '/data/test-tpf-star.fits'
```
Comment (barentsen) [inline lightkurve/tests/test_search.py]: I suggest replacing `os.path.dirname(os.path.abspath(__file__))` with `PACKAGEDIR`  (which is defined in `__init__.py`, i.e. you can do `from . import PACKAGEDIR`)

**Comment by nksaunders** [comment]
This looks good to me! @barentsen if you're happy with it, go ahead and merge.

**Comment by barentsen** [comment]
Unit tests are green!  Merging in 3 ... 2 ...

**Pull request merged**